lib/chipmunk_bag.rb

```diff
@@ -1,17 +1,41 @@
 require 'bagit'
+require 'net/http'
 
 class ChipmunkBag < BagIt::Bag
   def chipmunk_info_txt_file
     File.join bag_dir, 'chipmunk-info.txt'
   end
   
   def chipmunk_info
+    return {} unless File.exists?(chipmunk_info_txt_file)
     read_info_file chipmunk_info_txt_file
   end
 
   def write_chipmunk_info(hash)
     write_info_file chipmunk_info_txt_file, hash
     add_tag_file('chipmunk-info.txt')
   end
+
+  def download_metadata
+    File.write(File.join(bag_dir,chipmunk_info['Metadata-Tagfile']),
+      Net::HTTP.get(URI(chipmunk_info['Metadata-URL'])))
+
+    add_tag_file(chipmunk_info['Metadata-Tagfile'])
+  end
+
+  # Moves a file from the source location into the bag.
+  # 
+  # dest_path should be relative to the bag's data directory; parent
+  # directories will be created if they do not exist.
+  def add_file_by_moving(base_path,src_path)
```

Comment:
Looking more into `BagIt::Bag#add_file`, I've pulled together why this is confusing me so much...

They are using `base_path` to mean almost exactly opposite what I think of as a base path -- that is, the `data_dir` is what I would call the base path for the bag, and the `base_path` is the relative path within it.

The names `dest_path` or `rel_path` make so much more sense to me here -- a relative, full path to the file _within_ in the bag. And `src_path` seems right -- an absolute path (or, scary, maybe path relative to app pwd) to the file to place there.

So, there are two options...

1. Use `base_path` to maintain consistency with Bag (with a comment that it is relative to the bag's `data_dir`)
1. Use `dest_path` to avoid inconsistency with normal usage of `base_path` (which is a positional parameter, so the caller could use whatever is desired; noting that Bag calls this `base_path`).

I would prefer to use option 2 and a pull request to tipr/bagit to clean up those semantics. These are an artifact of the Ruby implementation. In fact, the spec is quite clear that the "base directory" is the top level (called `bag_dir` in BagIt::Bag). https://tools.ietf.org/html/draft-kunze-bagit-08#section-2

Reply:
Opened a pull request on tipr/bagit: https://github.com/tipr/bagit/pull/32